.werft/platform-delete-preview-environments-cron.ts

```diff
@@ -37,38 +39,92 @@ async function deletePreviewEnvironments() {
     }
     werft.done("prep")
 
-    werft.phase("Fetching branches");
-    const branches = getAllBranches();
-    // During the transition from the old preview names to the new ones we have to check for the existence of both the old or new
-    // preview name patterns before it is safe to delete a namespace.
-    const expectedPreviewEnvironmentNamespaces = new Set(branches.flatMap(branch => [parseBranch(branch), expectedNamespaceFromBranch(branch)]));
-    werft.done("Fetching branches");
-
     werft.phase("Fetching previews");
     let previews: string[]
     try {
         previews = listAllPreviewNamespaces(CORE_DEV_KUBECONFIG_PATH, {});
-        previews.forEach(previewNs => werft.log("Fetching previews", previewNs))
     } catch (err) {
         werft.fail("Fetching previews", err)
     }
     werft.done("Fetching previews");
 
+    werft.phase("Fetching outdated branches");
+    const branches = getAllBranches();
+    const outdatedPreviews = [];
+    // During the transition from the old preview names to the new ones we have to check for the existence of both the old or new
+    // preview name patterns before it is safe to delete a namespace.
+    branches.forEach(branch => {
+        const lastCommit = exec(`git log ${branch} --since=$(date +%Y-%m-%d -d "5 days ago")`, { silent: true })
+        if ( lastCommit.length < 1 ) {
+            const outdatedPreview = parseBranch(branch)
+            outdatedPreviews.push(outdatedPreview.trim())
+        }
+    })
+    const expectedPreviewEnvironmentNamespaces = new Set(branches.flatMap(branch => [parseBranch(branch), expectedNamespaceFromBranch(branch)]));
+
+    werft.phase("checking activity")
+    previews.forEach(previewNS => {
+        checkActivity(previewNS);
+    })
+
     werft.phase("deleting previews")
     try {
-        const previewsToDelete = previews.filter(ns => !expectedPreviewEnvironmentNamespaces.has(ns))
+        previewsToDelete = previewsToDelete.concat(previews.filter(ns => !expectedPreviewEnvironmentNamespaces.has(ns)))
+        previewsToDelete = previewsToDelete.concat(previews.filter(ns => new Set(outdatedPreviews).has(ns)))
+        previewsToDelete.forEach(previewNs => werft.log("Deleting previews", previewNs))
         // Trigger namespace deletion in parallel
-        const promises = previewsToDelete.map(preview => wipePreviewEnvironmentAndNamespace(helmInstallName, preview, CORE_DEV_KUBECONFIG_PATH, { slice: `Deleting preview ${preview}` }));
-        // But wait for all of them to finish before (or one of them to fail) before we continue
-        await Promise.all(promises)
+        // const promises = previewsToDelete.map(preview => wipePreviewEnvironmentAndNamespace(helmInstallName, preview, CORE_DEV_KUBECONFIG_PATH, { slice: `Deleting preview ${preview}` }));
+        // // But wait for all of them to finish before (or one of them to fail) before we continue
+        // await Promise.all(promises)
```

Comment:
This function needs to be enabled to actually remove preview-environments.

Reply:
Instead of commenting this out I'd prefer if you could implement a dry-run mechanisms for this job. If dry-run is enabled it should just echo "Would have deleted preview environment XXX" instead of actually doing the deletion.

Reply:
OK